orca-echo/src/test/groovy/com/netflix/spinnaker/orca/echo/pipeline/ManualJudgmentStageSpec.groovy

```diff
@@ -27,31 +30,67 @@ import static com.netflix.spinnaker.orca.echo.pipeline.ManualJudgmentStage.Notif
 import static com.netflix.spinnaker.orca.echo.pipeline.ManualJudgmentStage.WaitForManualJudgmentTask
 
 class ManualJudgmentStageSpec extends Specification {
+
+  EchoService echoService = Mock(EchoService)
+
+  FiatPermissionEvaluator fpe = Mock(FiatPermissionEvaluator)
+
   @Unroll
   void "should return execution status based on judgmentStatus"() {
     given:
-    def task = new WaitForManualJudgmentTask()
+    def task = new WaitForManualJudgmentTask(Optional.of(echoService), Optional.of(fpe))
 
     when:
     def result = task.execute(new StageExecutionImpl(PipelineExecutionImpl.newPipeline("orca"), "", context))
 
     then:
     result.status == expectedStatus
-    result.context.isEmpty()
 
     where:
     context                      || expectedStatus
     [:]                          || ExecutionStatus.RUNNING
-    [judgmentStatus: "continue"] || ExecutionStatus.SUCCEEDED
-    [judgmentStatus: "Continue"] || ExecutionStatus.SUCCEEDED
-    [judgmentStatus: "stop"]     || ExecutionStatus.TERMINAL
-    [judgmentStatus: "STOP"]     || ExecutionStatus.TERMINAL
-    [judgmentStatus: "unknown"]  || ExecutionStatus.RUNNING
+    [judgmentStatus: "continue", isAuthorized: true] || ExecutionStatus.SUCCEEDED
+    [judgmentStatus: "Continue", isAuthorized: true] || ExecutionStatus.SUCCEEDED
+    [judgmentStatus: "stop", isAuthorized: true]     || ExecutionStatus.TERMINAL
+    [judgmentStatus: "STOP", isAuthorized: true]     || ExecutionStatus.TERMINAL
+    [judgmentStatus: "unknown", isAuthorized: true]  || ExecutionStatus.RUNNING
+  }
+
+  @Unroll
+  void "should return execution status based on authorizedGroups"() {
+    given:
+    def task = new WaitForManualJudgmentTask(Optional.of(echoService), Optional.of(fpe))
+
+    when:
+    def stage = new StageExecutionImpl(PipelineExecutionImpl.newPipeline("orca"), "", context)
+    stage.lastModified = new StageExecution.LastModifiedDetails(user: "abc@somedomain.io", allowedAccounts: ["group1"])
+    def result = task.execute(stage)
+
+    then:
+    1 * fpe.getPermission('abc@somedomain.io') >> {
+      new UserPermission().addResources([new Role('foo'), new Role('baz')]).view
+    }
+    result.status == expectedStatus
+
+    where:
+    context                      || expectedStatus
+    [judgmentStatus: "continue", isAuthorized: false, stageRoles: ['foo'],
+     permissions: [WRITE: ["foo"],READ: ["foo","baz"], EXECUTE: ["foo"]]] || ExecutionStatus.SUCCEEDED
```

Comment:
What is the difference between `permissions` and `stageRoles`? I thought the intention was you needed to have a particular role (or roles?). 

The reason for WRITE/READ/EXECUTE isn't clear.

Reply:
stageRoles --> Roles configured at  the manual judgment stage.
permissions --> roles configured during the application.creation.

Please refer the design document table for more clear picture.
https://github.com/spinnaker/governance/pull/80

Reply:
> Display all the groups the user has selected while creating application(permissions) in the authorized groups dropdown.

Sure. But that set of application permissions should just be looked up from the UI, it doesn't need to be stored in the stage context. 

I'd like this PR to really be centered around the managed delivery stage and required user roles needed to act on it.

Reply:
I am unable to follow you here. Did not understand what you meant? Can you be more specific.

Reply:
Sorry, for some odd reason I typed "managed delivery" when I meant "manual judgment". I blame the late hour, sorry :(

I think what would be clearer is if I did a quick prototype/pr of what is in my mind, and we can compare. I'll do that this morning.